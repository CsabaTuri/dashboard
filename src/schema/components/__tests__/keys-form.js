import React from "react"
import Immutable from "immutable"
import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"

import { KeysForm } from "../keys-form"

jest.mock("../../../notifications", () => ({
  notify: (msg, fn) => fn()
}))

describe("KeysForm Component", () => {
  let comp, client

  beforeEach(() => {
    client = {
      query: jest.fn(() => Promise.resolve({
        secret: "one-time-secret"
      }))
    }

    comp = shallow(
      <KeysForm
        client={client}
        database={Immutable.fromJS({
          databases: [
            { name: "db1" },
            { name: "db2" }
          ]
        })} />
    )
  })

  it("should render an empty form", () => {
    expect(shallowToJson(comp)).toMatchSnapshot()
  })

  it("should create a new key on submit", () => {
    comp.find("TextField[label='Name']").simulate("beforeChange", "server-key")
    comp.find("WithResponsiveMode[label='Role']").simulate("changed", { key: "admin" })
    comp.find("WithResponsiveMode[label='Database']").simulate("changed", { key: "db1" })
    expect(shallowToJson(comp)).toMatchSnapshot()

    return comp.find("Connect(SchemaForm)").props().onSubmit().then(() => {
      comp.find("Connect(SchemaForm)").simulate("finish")
      expect(shallowToJson(comp)).toMatchSnapshot() // Form clean + secret being displayed
    })
  })
})
