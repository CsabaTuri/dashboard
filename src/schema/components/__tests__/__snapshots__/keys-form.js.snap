// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeysForm Component should be submit the form 1`] = `
<div>
  <Connect(SchemaForm)
    buttonText="Create Key"
    onFinish={[Function]}
    onSubmit={[Function]}
    title="Create a new key"
  >
    <TextField
      deferredValidationTime={200}
      errorMessage=""
      label="Name"
      multiline={false}
      onBeforeChange={[Function]}
      onChanged={[Function]}
      onGetErrorMessage={[Function]}
      onNotifyValidationResult={[Function]}
      resizable={true}
      underlined={false}
      value="server-key"
    />
    <Dropdown
      label="Role"
      onChanged={[Function]}
      options={
        Array [
          Object {
            "key": "admin",
            "text": "admin",
          },
          Object {
            "key": "server",
            "text": "server",
          },
          Object {
            "key": "server-readonly",
            "text": "server-readonly",
          },
          Object {
            "key": "client",
            "text": "client",
          },
        ]
      }
      selectedKey="admin"
    />
    <p
      className="ms-TextField-description"
    >
      Check 
      <a
        href="https://fauna.com/documentation/#authentication"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://fauna.com/documentation/#authentication
      </a>
       for more information about the key's role.
    </p>
    <Dropdown
      label="Database"
      onChanged={[Function]}
      options={
        Array [
          Object {
            "key": "db1",
            "text": "db1",
          },
          Object {
            "key": "db2",
            "text": "db2",
          },
        ]
      }
      selectedKey="db1"
    />
  </Connect(SchemaForm)>
</div>
`;

exports[`KeysForm Component should be submit the form 2`] = `
<div>
  <Connect(SchemaForm)
    buttonText="Create Key"
    onFinish={[Function]}
    onSubmit={[Function]}
    title="Create a new key"
  >
    <TextField
      deferredValidationTime={200}
      errorMessage=""
      label="Name"
      multiline={false}
      onBeforeChange={[Function]}
      onChanged={[Function]}
      onGetErrorMessage={[Function]}
      onNotifyValidationResult={[Function]}
      resizable={true}
      underlined={false}
      value=""
    />
    <Dropdown
      label="Role"
      onChanged={[Function]}
      options={
        Array [
          Object {
            "key": "admin",
            "text": "admin",
          },
          Object {
            "key": "server",
            "text": "server",
          },
          Object {
            "key": "server-readonly",
            "text": "server-readonly",
          },
          Object {
            "key": "client",
            "text": "client",
          },
        ]
      }
      selectedKey={null}
    />
    <p
      className="ms-TextField-description"
    >
      Check 
      <a
        href="https://fauna.com/documentation/#authentication"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://fauna.com/documentation/#authentication
      </a>
       for more information about the key's role.
    </p>
    <Dropdown
      label="Database"
      onChanged={[Function]}
      options={
        Array [
          Object {
            "key": "db1",
            "text": "db1",
          },
          Object {
            "key": "db2",
            "text": "db2",
          },
        ]
      }
      selectedKey={null}
    />
  </Connect(SchemaForm)>
</div>
`;

exports[`KeysForm Component should create a new key on submit 1`] = `
<div>
  <Connect(SchemaForm)
    buttonText="Create Key"
    onFinish={[Function]}
    onSubmit={[Function]}
    title="Create a new key"
  >
    <TextField
      deferredValidationTime={200}
      errorMessage=""
      label="Name"
      multiline={false}
      onBeforeChange={[Function]}
      onChanged={[Function]}
      onGetErrorMessage={[Function]}
      onNotifyValidationResult={[Function]}
      resizable={true}
      underlined={false}
      value=""
    />
    <Dropdown
      label="Role"
      onChanged={[Function]}
      options={
        Array [
          Object {
            "key": "admin",
            "text": "admin",
          },
          Object {
            "key": "server",
            "text": "server",
          },
          Object {
            "key": "server-readonly",
            "text": "server-readonly",
          },
          Object {
            "key": "client",
            "text": "client",
          },
        ]
      }
      selectedKey={null}
    />
    <p
      className="ms-TextField-description"
    >
      Check 
      <a
        href="https://fauna.com/documentation/#authentication"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://fauna.com/documentation/#authentication
      </a>
       for more information about the key's role.
    </p>
    <Dropdown
      label="Database"
      onChanged={[Function]}
      options={
        Array [
          Object {
            "key": "db1",
            "text": "db1",
          },
          Object {
            "key": "db2",
            "text": "db2",
          },
        ]
      }
      selectedKey={null}
    />
  </Connect(SchemaForm)>
  <div>
    <br />
    <MessageBar
      isMultiline={true}
      messageBarType={3}
      onDismiss={null}
    >
      Your key's secret is: 
      <pre>
        one-time-secret
      </pre>
      Be aware that this secret won't be displayed again.
    </MessageBar>
  </div>
</div>
`;

exports[`KeysForm Component should render an empty form 1`] = `
<div>
  <Connect(SchemaForm)
    buttonText="Create Key"
    onFinish={[Function]}
    onSubmit={[Function]}
    title="Create a new key"
  >
    <TextField
      deferredValidationTime={200}
      errorMessage=""
      label="Name"
      multiline={false}
      onBeforeChange={[Function]}
      onChanged={[Function]}
      onGetErrorMessage={[Function]}
      onNotifyValidationResult={[Function]}
      resizable={true}
      underlined={false}
      value=""
    />
    <Dropdown
      label="Role"
      onChanged={[Function]}
      options={
        Array [
          Object {
            "key": "admin",
            "text": "admin",
          },
          Object {
            "key": "server",
            "text": "server",
          },
          Object {
            "key": "server-readonly",
            "text": "server-readonly",
          },
          Object {
            "key": "client",
            "text": "client",
          },
        ]
      }
      selectedKey={null}
    />
    <p
      className="ms-TextField-description"
    >
      Check 
      <a
        href="https://fauna.com/documentation/#authentication"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://fauna.com/documentation/#authentication
      </a>
       for more information about the key's role.
    </p>
    <Dropdown
      label="Database"
      onChanged={[Function]}
      options={
        Array [
          Object {
            "key": "db1",
            "text": "db1",
          },
          Object {
            "key": "db2",
            "text": "db2",
          },
        ]
      }
      selectedKey={null}
    />
  </Connect(SchemaForm)>
</div>
`;
